### PR TITLE
Lra 44 setup required fields in tables

### PR DESCRIPTION
**Pull request by britaumich**
This is not a pull request yet because the Infrastructure group hasn't added required fields to the [It Security Incidents] tab in the [Security Tracking Tool - Database fields] spreadsheet. But I want you to review the code.

### CODE REVIEW

**Review by rsmoke — COMMENTED** [review]
This is very nice. I especially like the custom validator

**Comment by britaumich** [comment]
I merged LRA-42-device-creation-for-main-tables to this branch because I had to change the database tables (and hence views) in this branch and therefore couldn't work in LRA-42-device-creation-for-main-tables branch (tables and view didn't match).
The following I added to the device-create function:
- search the TDX assets database by serial number and hostname (user enter a serial or hostname into [Add device] field)
- error message in case multiple records are returned by TDX API
- create a device with hostname or serial

**Comment by britaumich** [comment]
I improved the TDX API method (get_device_data) a lot:
- moved it to a module
- call it from controllers: legacy_os_record create and show and device create methods
Added incomplete scope and incomplete? method to the device model  (not sure if queries are correct)

File: app/models/sensitive_data_system.rb [hunk 1]
```diff
@@ -3,17 +3,17 @@
...
 #  id                                  :bigint           not null, primary key
-#  owner_username                      :string(255)
+#  owner_username                      :string(255)      not null
 #  owner_full_name                     :string(255)
 #  dept                                :string(255)
 #  phone                               :string(255)
 #  additional_dept_contact             :string(255)
 #  additional_dept_contact_phone       :string(255)
-#  support_poc                         :string(255)
+#  support_poc                         :string(255)      not null
```
Comment (rsmoke) [inline app/models/sensitive_data_system.rb]: what does poc mean???

**Review by rsmoke — COMMENTED** [review]
This is pretty

**Pull request closed without merging**